From patchwork Tue Feb 12 23:34:37 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808905
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id E56EF13A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:54 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id D51F82BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:54 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id C97152C201; Tue, 12 Feb 2019 23:34:54 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 693412BED9
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:54 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731272AbfBLXex (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:34:53 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1730661AbfBLXex (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:53 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:52 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494776"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:52 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 01/15] KVM: Add a "-t" option to run a specific
 test
Date: Tue, 12 Feb 2019 15:34:37 -0800
Message-Id: <20190212233451.27740-2-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

x86's nested-VMX test space is massive, to the point where the group
option, "-g", does not provide sufficient granularity, e.g. a user may
want to run the core VMX tests but not the VMCS shadow tests, which take
an absurdly long time to run due to exhuastively testing all VMCS field
values.

Note that x86's config is currently a bit of a train wreck as far as VMX
is concerned; the "-t" option is of limited value until the config is
revamped.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 run_tests.sh         | 6 +++++-
 scripts/runtime.bash | 6 +++---
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/run_tests.sh b/run_tests.sh
index aa2e65f..7d55e74 100755
--- a/run_tests.sh
+++ b/run_tests.sh
@@ -21,6 +21,7 @@ Usage: $0 [-h] [-v] [-a] [-g group] [-j NUM-TASKS]
     -a: Run all tests, including those flagged as 'nodefault'
         and those guarded by errata.
     -g: Only execute tests in the given group
+    -t: Only execute the given test
     -j: Execute tests in parallel
 
 Set the environment variable QEMU=/path/to/qemu-system-ARCH to
@@ -32,7 +33,7 @@ EOF
 RUNTIME_arch_run="./$TEST_DIR/run"
 source scripts/runtime.bash
 
-while getopts "ag:hj:v" opt; do
+while getopts "ag:t:hj:v" opt; do
     case $opt in
         a)
             run_all_tests="yes"
@@ -41,6 +42,9 @@ while getopts "ag:hj:v" opt; do
         g)
             only_group=$OPTARG
             ;;
+        t)
+            only_test=$OPTARG
+            ;;
         h)
             usage
             exit
diff --git a/scripts/runtime.bash b/scripts/runtime.bash
index 3f5f3e2..c2d0b6b 100644
--- a/scripts/runtime.bash
+++ b/scripts/runtime.bash
@@ -65,9 +65,9 @@ function run()
 
     if [ -z "$testname" ]; then
         return
-    fi
-
-    if [ -n "$only_group" ] && ! grep -qw "$only_group" <<<$groups; then
+    elif [ -n "$only_test" ] && [ "$only_test" != "$testname" ]; then
+        return
+    elif [ -n "$only_group" ] && ! grep -qw "$only_group" <<<$groups; then
         return
     fi
 

From patchwork Tue Feb 12 23:34:38 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808907
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id B278E13A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:55 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A1D962BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:55 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 967282C201; Tue, 12 Feb 2019 23:34:55 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 261A12BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:55 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731620AbfBLXey (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:34:54 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1729589AbfBLXex (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:53 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:53 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494778"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:53 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 02/15] KVM: x86/config: Consolidate EPT access
 tests into a single test
Date: Tue, 12 Feb 2019 15:34:38 -0800
Message-Id: <20190212233451.27740-3-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Defining individual testcases in the config for each individual test in
vmx.flat has proven to be a maintenance nightmare, e.g. has led to poor
test configurations due to naive copy+paste.  For example, many VMX
tests specify "-m 2560" or "-m 2048" even though only the EPT access
tests actually require >2gb of memory.  Take advantage of vmx.flat's
simple glob parsing to create a single testcase for the EPC access
tests.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
Reviewed-by: Liam Merwick <liam.merwick@oracle.com>
---
 x86/unittests.cfg | 126 +++-------------------------------------------
 1 file changed, 6 insertions(+), 120 deletions(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 64ecb43..3f23cde 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -230,6 +230,12 @@ extra_params = -cpu host,+vmx -m 2560 -append -exit_monitor_from_l2_test
 arch = x86_64
 groups = vmx
 
+[ept]
+file = vmx.flat
+extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append "ept_access*"
+arch = x86_64
+groups = vmx
+
 [vmx_null]
 file = vmx.flat
 extra_params = -cpu host,+vmx -append null
@@ -410,126 +416,6 @@ extra_params = -cpu host,+vmx -append "v2_null_test v2_multiple_entries_test fix
 arch = x86_64
 groups = vmx
 
-[vmx_ept_access_test_not_present]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_not_present
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_read_only]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_read_only
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_write_only]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_write_only
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_read_write]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_read_write
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_execute_only]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_execute_only
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_read_execute]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_read_execute
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_write_execute]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_write_execute
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_read_write_execute]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_read_write_execute
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_reserved_bits]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_reserved_bits
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_ignored_bits]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_ignored_bits
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_paddr_not_present_ad_disabled]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_paddr_not_present_ad_disabled
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_paddr_not_present_ad_enabled]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_paddr_not_present_ad_enabled
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_paddr_read_only_ad_disabled]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_paddr_read_only_ad_disabled
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_paddr_read_only_ad_enabled]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_paddr_read_only_ad_enabled
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_paddr_read_write]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_paddr_read_write
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_paddr_read_write_execute]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_paddr_read_write_execute
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_paddr_read_execute_ad_disabled]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_paddr_read_execute_ad_disabled
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_paddr_read_execute_ad_enabled]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_paddr_read_execute_ad_enabled
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_paddr_not_present_page_fault]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_paddr_not_present_page_fault
-arch = x86_64
-groups = vmx
-
-[vmx_ept_access_test_force_2m_page]
-file = vmx.flat
-extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append ept_access_test_force_2m_page
-arch = x86_64
-groups = vmx
-
 [vmx_invvpid]
 file = vmx.flat
 extra_params = -cpu host,+vmx -m 2560 -append invvpid_test_v2

From patchwork Tue Feb 12 23:34:39 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808933
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id B96CE1390
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:14 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A70122BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:14 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 9B9402C20C; Tue, 12 Feb 2019 23:35:14 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 52F062BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:14 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731796AbfBLXfN (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:35:13 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1730661AbfBLXex (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:53 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:53 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494780"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:53 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 03/15] KVM: nVMX: Exclude the EPT access tests
 from the VMX testcase
Date: Tue, 12 Feb 2019 15:34:39 -0800
Message-Id: <20190212233451.27740-4-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Due to unique requirements, the EPT access tests have a dedicated
testcase.  Remove them from the primary VMX testcase so that future
patches can make the primary testcase less needy, e.g. drop the 2gb
memory requirement.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 3f23cde..08b9a7e 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -226,7 +226,7 @@ extra_params = -cpu qemu64,+umip
 
 [vmx]
 file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append -exit_monitor_from_l2_test
+extra_params = -cpu host,+vmx -m 2560 -append "-exit_monitor_from_l2_test -ept_access*"
 arch = x86_64
 groups = vmx
 

From patchwork Tue Feb 12 23:34:40 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808931
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id F41FA1390
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:12 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id E3CBF2BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:12 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id D857B2C20C; Tue, 12 Feb 2019 23:35:12 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 920282BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:12 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731941AbfBLXfL (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:35:11 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731796AbfBLXey (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:54 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:53 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494781"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:53 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 04/15] KVM: nVMX: Drop the bogus 2gb
 requirement from the "vmx" testcase
Date: Tue, 12 Feb 2019 15:34:40 -0800
Message-Id: <20190212233451.27740-5-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

...now that it excludes the EPT access tests, which are the only tests
that actually need >2gb of memory.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 08b9a7e..ba1c409 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -226,7 +226,7 @@ extra_params = -cpu qemu64,+umip
 
 [vmx]
 file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append "-exit_monitor_from_l2_test -ept_access*"
+extra_params = -cpu host,+vmx -append "-exit_monitor_from_l2_test -ept_access*"
 arch = x86_64
 groups = vmx
 

From patchwork Tue Feb 12 23:34:41 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808925
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 542261390
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:06 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 4325C2BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:06 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 36BC42C201; Tue, 12 Feb 2019 23:35:06 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id D91472BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:05 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731822AbfBLXey (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:34:54 -0500
Received: from mga04.intel.com ([192.55.52.120]:58482 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731804AbfBLXey (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:54 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:53 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494782"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:53 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 05/15] KVM: nVMX: Drop the bogus 2gb
 requirement from the shadow VMCS test
Date: Tue, 12 Feb 2019 15:34:41 -0800
Message-Id: <20190212233451.27740-6-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The shadow VMCS tests don't have any special memory requirements, this
appears to be a copy+paste error.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index ba1c409..e24821f 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -519,7 +519,7 @@ groups = vmx
 [vmx_vmcs_shadow_test]
 file = vmx.flat
 smp = 2
-extra_params = -cpu host,+vmx -m 2048 -append vmx_vmcs_shadow_test
+extra_params = -cpu host,+vmx -append vmx_vmcs_shadow_test
 arch = x86_64
 groups = vmx
 

From patchwork Tue Feb 12 23:34:42 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808909
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 68D6A13A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:56 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 58E0B2BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:56 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 4D63E2C201; Tue, 12 Feb 2019 23:34:56 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id F34CE2BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:55 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731852AbfBLXez (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:34:55 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1729589AbfBLXey (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:54 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:53 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494783"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:53 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 06/15] KVM: nVMX: Drop the bogus SMP
 requirement from VMCS shadowing test
Date: Tue, 12 Feb 2019 15:34:42 -0800
Message-Id: <20190212233451.27740-7-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The VMCS shadowing tests are purely single threaded.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 1 -
 1 file changed, 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index e24821f..af26170 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -518,7 +518,6 @@ groups = vmx
 
 [vmx_vmcs_shadow_test]
 file = vmx.flat
-smp = 2
 extra_params = -cpu host,+vmx -append vmx_vmcs_shadow_test
 arch = x86_64
 groups = vmx

From patchwork Tue Feb 12 23:34:43 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808929
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 8B65A17FB
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:08 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 7BCD72BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:08 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 701C52C20C; Tue, 12 Feb 2019 23:35:08 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 1B6222BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:08 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1732032AbfBLXfH (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:35:07 -0500
Received: from mga04.intel.com ([192.55.52.120]:58482 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731808AbfBLXey (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:54 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:54 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494784"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:53 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 07/15] KVM: nVMX: Rename VMCS shadowing test to
 "vmcs_shadow"
Date: Tue, 12 Feb 2019 15:34:43 -0800
Message-Id: <20190212233451.27740-8-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Now that the run script has a "-t" option to run a specific test, use a
name that requires less typing.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index af26170..7c0310c 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -516,7 +516,7 @@ extra_params = -cpu host,+vmx -m 2048 -append vmx_apic_passthrough_thread_test
 arch = x86_64
 groups = vmx
 
-[vmx_vmcs_shadow_test]
+[vmcs_shadow]
 file = vmx.flat
 extra_params = -cpu host,+vmx -append vmx_vmcs_shadow_test
 arch = x86_64

From patchwork Tue Feb 12 23:34:44 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808927
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 10F8413A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:08 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id F13162BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:07 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id E36012C20C; Tue, 12 Feb 2019 23:35:07 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 6751C2BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:07 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1732007AbfBLXfG (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:35:06 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731817AbfBLXey (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:54 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:54 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494785"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:54 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 08/15] KVM: nVMX: Exclude the VMCS shadowing
 test from the "vmx" testcase
Date: Tue, 12 Feb 2019 15:34:44 -0800
Message-Id: <20190212233451.27740-9-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Due to exhaustively probing all possible VMCS field values, the VMCS
shadowing test takes forever and a day to run, to the point where it
times out when running the unit tests in L1.  Now that VMCS shadowing
has its own sane testcase, exclude it from the generic VMX testcase.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 7c0310c..ad54126 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -226,7 +226,7 @@ extra_params = -cpu qemu64,+umip
 
 [vmx]
 file = vmx.flat
-extra_params = -cpu host,+vmx -append "-exit_monitor_from_l2_test -ept_access*"
+extra_params = -cpu host,+vmx -append "-exit_monitor_from_l2_test -ept_access* -vmx_vmcs_shadow_test"
 arch = x86_64
 groups = vmx
 

From patchwork Tue Feb 12 23:34:45 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808917
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id E2F1817FB
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:00 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id D0C432BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:00 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id C54482C201; Tue, 12 Feb 2019 23:35:00 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 39C002BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:00 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731853AbfBLXe5 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:34:57 -0500
Received: from mga04.intel.com ([192.55.52.120]:58482 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731819AbfBLXe4 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:56 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:54 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494787"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:54 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 09/15] KVM: nVMX: Drop testcases that are
 redundant with the primary VMX testcase
Date: Tue, 12 Feb 2019 15:34:45 -0800
Message-Id: <20190212233451.27740-10-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The overarching VMX test binary, i.e. vmx.flat, contains a large number
of individual tests.  In addition to a kitchen sink testcase, "vmx",
many (but not all) of the individual tests also define their own test
case.  Drop all such individual test cases that can use the generic
umbrella testcase, e.g. don't need SMP, 2gb ram, etc..., so that running
all unit tests doesn't double up on said tests.

Alternatively, "vmx" could be removed, but doing so would provide little
value (the tests are relatively fast and stable), be a maintenance
nightmare, and run the risk of unintentionally omitting tests from the
config.

Ideally, the config would find a middle ground and group the vmx tests
into a handful of meaningful testcases, but the tests exposed by
vmx.flat currently don't play nice with that approach.  In other words,
the tests themselves need to be reorganized before the config can follow
suit.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 252 ----------------------------------------------
 1 file changed, 252 deletions(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index ad54126..4d46789 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -236,258 +236,6 @@ extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append "ept_access*"
 arch = x86_64
 groups = vmx
 
-[vmx_null]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append null
-arch = x86_64
-groups = vmx
-
-[vmx_test_vmx_feature_control]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append test_vmx_feature_control
-arch = x86_64
-groups = vmx
-
-[vmx_test_vmxon]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append test_vmxon
-arch = x86_64
-groups = vmx
-
-[vmx_test_vmptrld]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append test_vmptrld
-arch = x86_64
-groups = vmx
-
-[vmx_test_vmclear]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append test_vmclear
-arch = x86_64
-groups = vmx
-
-[vmx_test_vmptrst]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append test_vmptrst
-arch = x86_64
-groups = vmx
-
-[vmx_test_vmwrite_vmread]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append test_vmwrite_vmread
-arch = x86_64
-groups = vmx
-
-[vmx_test_vmcs_high]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append test_vmcs_high
-arch = x86_64
-groups = vmx
-
-[vmx_test_vmcs_lifecycle]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append test_vmcs_lifecycle
-arch = x86_64
-groups = vmx
-
-[vmx_test_vmx_caps]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append test_vmx_caps
-arch = x86_64
-groups = vmx
-
-[vmx_vmenter]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append vmenter
-arch = x86_64
-groups = vmx
-
-[vmx_preemption_timer]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append preemption_timer
-arch = x86_64
-groups = vmx
-
-[vmx_control_field_PAT]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append control_field_PAT
-arch = x86_64
-groups = vmx
-
-[vmx_control_field_EFER]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append control_field_EFER
-arch = x86_64
-groups = vmx
-
-[vmx_CR_shadowing]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append CR_shadowing
-arch = x86_64
-groups = vmx
-
-[vmx_IO_bitmap]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append I/O_bitmap
-arch = x86_64
-groups = vmx
-
-[vmx_instruction_intercept]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append instruction_intercept
-arch = x86_64
-groups = vmx
-
-[vmx_EPT_AD_enabled]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append EPT_A/D_enabled
-arch = x86_64
-groups = vmx
-
-[vmx_EPT_AD_disabled]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append EPT_A/D_disabled
-arch = x86_64
-groups = vmx
-
-[vmx_PML]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append PML
-arch = x86_64
-groups = vmx
-
-[vmx_VPID]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append VPID
-arch = x86_64
-groups = vmx
-
-[vmx_interrupt]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append interrupt
-arch = x86_64
-groups = vmx
-
-[vmx_debug_controls]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append debug_controls
-arch = x86_64
-groups = vmx
-
-[vmx_MSR_switch]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append MSR_switch
-arch = x86_64
-groups = vmx
-
-[vmx_vmmcall]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append vmmcall
-arch = x86_64
-groups = vmx
-
-[vmx_disable_RDTSCP]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append disable_RDTSCP
-arch = x86_64
-groups = vmx
-
-[vmx_int3]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append int3
-arch = x86_64
-groups = vmx
-
-[vmx_into]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append into
-arch = x86_64
-groups = vmx
-
-[vmx_exit_monitor_from_l2_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append exit_monitor_from_l2_test
-arch = x86_64
-groups = vmx
-
-[vmx_v2]
-file = vmx.flat
-extra_params = -cpu host,+vmx -append "v2_null_test v2_multiple_entries_test fixture_test_case1 fixture_test_case2"
-arch = x86_64
-groups = vmx
-
-[vmx_invvpid]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append invvpid_test_v2
-arch = x86_64
-groups = vmx
-
-[vmx_controls]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_controls_test
-arch = x86_64
-groups = vmx
-
-[vmx_vmentry_movss_shadow_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmentry_movss_shadow_test
-arch = x86_64
-groups = vmx
-
-[vmx_cr_load_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_cr_load_test
-arch = x86_64
-groups = vmx
-
-[vmx_nm_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_nm_test
-arch = x86_64
-groups = vmx
-
-[vmx_pending_event_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_pending_event_test
-arch = x86_64
-groups = vmx
-
-[vmx_pending_event_hlt_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_pending_event_hlt_test
-arch = x86_64
-groups = vmx
-
-[vmx_store_tsc_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_store_tsc_test
-arch = x86_64
-groups = vmx
-
-[vmx_store_tsc_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_store_tsc_test
-arch = x86_64
-groups = vmx
-
-[vmx_db_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_db_test
-arch = x86_64
-groups = vmx
-
-[vmx_nmi_window_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_nmi_window_test
-arch = x86_64
-groups = vmx
-
-[vmx_intr_window_test]
-file = vmx.flat
-extra_params = -cpu host,+vmx -m 2560 -append vmx_intr_window_test
-arch = x86_64
-groups = vmx
-
 [vmx_eoi_bitmap_ioapic_scan]
 file = vmx.flat
 smp = 2

From patchwork Tue Feb 12 23:34:46 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808915
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id B9E3813A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:00 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A831C2BED9
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:00 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 9CC592C202; Tue, 12 Feb 2019 23:35:00 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 5A95F2BED9
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:00 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731886AbfBLXez (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:34:55 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731804AbfBLXey (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:54 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:54 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494789"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:54 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 10/15] KVM: nVMX: Drop the SMP configuration of
 the ST APIC passthrough test
Date: Tue, 12 Feb 2019 15:34:46 -0800
Message-Id: <20190212233451.27740-11-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The APIC passthrough test has single-thread and dual-threaded variants.
Drop the testcase that runs the single-threaded variant with smp=2, it
provides no additional value over running the test in the generic "vmx"
testcase.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 4d46789..8839644 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -250,13 +250,6 @@ arch = x86_64
 groups = vmx
 timeout = 10
 
-[vmx_apic_passthrough]
-file = vmx.flat
-smp = 2
-extra_params = -cpu host,+vmx -m 2048 -append vmx_apic_passthrough_test
-arch = x86_64
-groups = vmx
-
 [vmx_apic_passthrough_thread]
 file = vmx.flat
 smp = 2

From patchwork Tue Feb 12 23:34:47 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808911
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id BF1641390
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:57 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id AF87D2BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:57 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id A41882C201; Tue, 12 Feb 2019 23:34:57 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 564122BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:57 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731899AbfBLXe4 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:34:56 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731853AbfBLXez (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:55 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:54 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494790"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:54 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 11/15] KVM: nVMX: Consolidate the SMP tests
 into a single testcase
Date: Tue, 12 Feb 2019 15:34:47 -0800
Message-Id: <20190212233451.27740-12-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The SMP tests aren't so special that each test deserves its own
testcase.  This also sets the stage for renaming the tests to add an
"smp_" prefix so that they can be run using simply globbing.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 8839644..838b2f5 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -236,10 +236,10 @@ extra_params = -cpu host,host-phys-bits,+vmx -m 2560 -append "ept_access*"
 arch = x86_64
 groups = vmx
 
-[vmx_eoi_bitmap_ioapic_scan]
+[vmx_smp]
 file = vmx.flat
 smp = 2
-extra_params = -cpu host,+vmx -m 2048 -append vmx_eoi_bitmap_ioapic_scan_test
+extra_params = -cpu host,+vmx -m 2048 -append "vmx_eoi_bitmap_ioapic_scan_test vmx_apic_passthrough_thread_test"
 arch = x86_64
 groups = vmx
 
@@ -250,13 +250,6 @@ arch = x86_64
 groups = vmx
 timeout = 10
 
-[vmx_apic_passthrough_thread]
-file = vmx.flat
-smp = 2
-extra_params = -cpu host,+vmx -m 2048 -append vmx_apic_passthrough_thread_test
-arch = x86_64
-groups = vmx
-
 [vmcs_shadow]
 file = vmx.flat
 extra_params = -cpu host,+vmx -append vmx_vmcs_shadow_test

From patchwork Tue Feb 12 23:34:48 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808919
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 415A11390
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:02 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 30C882BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:02 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 2301D2C201; Tue, 12 Feb 2019 23:35:02 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id C27E52BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:01 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731955AbfBLXfB (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:35:01 -0500
Received: from mga04.intel.com ([192.55.52.120]:58484 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731858AbfBLXez (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:55 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:54 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494791"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:54 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 12/15] KVM: nVMX: Rename SMP tests to being
 with vmx_smp_*
Date: Tue, 12 Feb 2019 15:34:48 -0800
Message-Id: <20190212233451.27740-13-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

...so that they can be specified via globbing in the config.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 2 +-
 x86/vmx_tests.c   | 8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 838b2f5..cab5b62 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -239,7 +239,7 @@ groups = vmx
 [vmx_smp]
 file = vmx.flat
 smp = 2
-extra_params = -cpu host,+vmx -m 2048 -append "vmx_eoi_bitmap_ioapic_scan_test vmx_apic_passthrough_thread_test"
+extra_params = -cpu host,+vmx -m 2048 -append "vmx_smp*"
 arch = x86_64
 groups = vmx
 
diff --git a/x86/vmx_tests.c b/x86/vmx_tests.c
index 28bab81..b51ce4c 100644
--- a/x86/vmx_tests.c
+++ b/x86/vmx_tests.c
@@ -5910,7 +5910,7 @@ static void vmx_eoi_bitmap_ioapic_scan_test_guest(void)
 	asm volatile ("int $0x79");
 }
 
-static void vmx_eoi_bitmap_ioapic_scan_test(void)
+static void vmx_smp_eoi_bitmap_ioapic_scan_test(void)
 {
 	if (!cpu_has_apicv() || (cpu_count() < 2)) {
 		report_skip(__func__);
@@ -6064,7 +6064,7 @@ static void vmx_apic_passthrough_test(void)
 	vmx_apic_passthrough(false);
 }
 
-static void vmx_apic_passthrough_thread_test(void)
+static void vmx_smp_apic_passthrough_test(void)
 {
 	vmx_apic_passthrough(true);
 }
@@ -6423,11 +6423,11 @@ struct vmx_test vmx_tests[] = {
 	TEST(vmx_controls_test),
 	TEST(vmentry_movss_shadow_test),
 	/* APICv tests */
-	TEST(vmx_eoi_bitmap_ioapic_scan_test),
+	TEST(vmx_smp_eoi_bitmap_ioapic_scan_test),
 	TEST(vmx_hlt_with_rvi_test),
 	/* APIC pass-through tests */
 	TEST(vmx_apic_passthrough_test),
-	TEST(vmx_apic_passthrough_thread_test),
+	TEST(vmx_smp_apic_passthrough_test),
 	/* VMCS Shadowing tests */
 	TEST(vmx_vmcs_shadow_test),
 	/* Regression tests */

From patchwork Tue Feb 12 23:34:49 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808923
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 571CA13A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:05 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 466422BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:05 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 37EE72C201; Tue, 12 Feb 2019 23:35:05 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id D8CDF2BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:04 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731962AbfBLXfD (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:35:03 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1729589AbfBLXez (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:55 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:55 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494792"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:54 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 13/15] KVM: nVMX: Exclude the SMP tests from
 the primary "vmx" testcases
Date: Tue, 12 Feb 2019 15:34:49 -0800
Message-Id: <20190212233451.27740-14-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The "vmx" testcase runs single-threaded, i.e. the tests are skipped
anyways.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index cab5b62..21e6d56 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -226,7 +226,7 @@ extra_params = -cpu qemu64,+umip
 
 [vmx]
 file = vmx.flat
-extra_params = -cpu host,+vmx -append "-exit_monitor_from_l2_test -ept_access* -vmx_vmcs_shadow_test"
+extra_params = -cpu host,+vmx -append "-exit_monitor_from_l2_test -ept_access* -vmx_smp* -vmx_vmcs_shadow_test"
 arch = x86_64
 groups = vmx
 

From patchwork Tue Feb 12 23:34:50 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808921
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id F1BAC13A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:02 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id E11E82BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:02 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id D56102C201; Tue, 12 Feb 2019 23:35:02 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 930302BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:35:02 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731920AbfBLXfA (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:35:00 -0500
Received: from mga04.intel.com ([192.55.52.120]:58481 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731869AbfBLXez (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:55 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:55 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494793"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:55 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 14/15] KVM: nVMX: Drop the bogus 2gb
 requirement from the SMP tests
Date: Tue, 12 Feb 2019 15:34:50 -0800
Message-Id: <20190212233451.27740-15-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The SMP tests don't have special memory requirements, this appears to be
a copy+paste error.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 21e6d56..063e63c 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -239,7 +239,7 @@ groups = vmx
 [vmx_smp]
 file = vmx.flat
 smp = 2
-extra_params = -cpu host,+vmx -m 2048 -append "vmx_smp*"
+extra_params = -cpu host,+vmx -append "vmx_smp*"
 arch = x86_64
 groups = vmx
 

From patchwork Tue Feb 12 23:34:51 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sean Christopherson <sean.j.christopherson@intel.com>
X-Patchwork-Id: 10808913
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 5FDD11390
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:59 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 4FD492BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:59 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 447392C201; Tue, 12 Feb 2019 23:34:59 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 017ED2BCDD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 Feb 2019 23:34:59 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731864AbfBLXe6 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 Feb 2019 18:34:58 -0500
Received: from mga04.intel.com ([192.55.52.120]:58484 "EHLO mga04.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1731887AbfBLXez (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 Feb 2019 18:34:55 -0500
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga006.fm.intel.com ([10.253.24.20])
  by fmsmga104.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 12 Feb 2019 15:34:55 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,362,1544515200";
   d="scan'208";a="318494794"
Received: from sjchrist-coffee.jf.intel.com ([10.54.74.14])
  by fmsmga006.fm.intel.com with ESMTP; 12 Feb 2019 15:34:55 -0800
From: Sean Christopherson <sean.j.christopherson@intel.com>
To: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>
Cc: kvm@vger.kernel.org
Subject: [kvm-unit-tests PATCH 15/15] KVM: nVMX: Drop the bogus 2gb
 requirement from the HLT+RVI testcase
Date: Tue, 12 Feb 2019 15:34:51 -0800
Message-Id: <20190212233451.27740-16-sean.j.christopherson@intel.com>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20190212233451.27740-1-sean.j.christopherson@intel.com>
References: <20190212233451.27740-1-sean.j.christopherson@intel.com>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The HLT+RVI testcase doesn't have special memory requirements, this
appears to be a copy+paste error.

Signed-off-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 063e63c..61fc210 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -245,7 +245,7 @@ groups = vmx
 
 [vmx_hlt_with_rvi_test]
 file = vmx.flat
-extra_params = -cpu host,+vmx -m 2048 -append vmx_hlt_with_rvi_test
+extra_params = -cpu host,+vmx -append vmx_hlt_with_rvi_test
 arch = x86_64
 groups = vmx
 timeout = 10
